server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java

```diff
@@ -672,6 +671,66 @@ public void testUpdateToRoutedRequestByPlannerWhereOr() throws Exception {
     }
 
     @Test
+    public void test_temporal_to_be_replaced_by_operators_test() throws Exception {
+        // Case sensitive LIKE
+        execute("select 'TextToMatch' LIKE 'Te\\%tch'");
```

Comment:
You're probably aware of it, but just to make sure: If the expression is in the select list - instead of on a column in the where clause - it will more or less run the same logic as if a unit test on the scalar/operator implementation would. That's much cheaper to run

And for query tests it would be cheaper to use the `QueryTester` and create test cases like in https://github.com/crate/crate/blob/df549946d1774f2539e0ce1e6b7ef1bc60dbf298/server/src/test/java/io/crate/expression/predicate/NotPredicateTest.java#L60-L78

Reply:
This PR has unit tests which cover this issue (they fail without the change) and I used them to debug. Itest here is rather for "demo" purpose, to be 100% sure that statement doesn't trip somewhere else. I removed it now

Reply:
The c-style escape is happening on the literals, so when we use columns those contain the correct string as the c-style escape has been applied when inserting rows to the table, right @BaurzhanSakhariev ?